### PR TITLE
Use smaller types for term and nonterm costs.

### PR DESCRIPTION
**Pull request by ltratt**
In retrospect I chose ludicrously large types. Why would I want to give a terminal a cost of `u64::max_value()` or thereabouts? I have no idea. To be honest, I suspect that in general the scoring of terminals is likely to be in the range of 1--10, and probably towards the lower end of that range.

On that basis, this commit changes things to only allow terminals to be scored with a `u8`. This then makes it possible to reduce the score type of a nonterminal to a `u32`. A rule would need to contain over 16 million terminals with a maximum score to get close to overflowing this which seems, to put it mildly, rather unlikely.

All in all, this commit helps reduce memory usage significantly; perhaps not so much in cfgrammar itself, but in other code which uses the costs produced here. It also gives people a clearer indication about the likely values they're likely to encounter in real life.

### CODE REVIEW

File: src/lib/yacc/grammar.rs [hunk 1]
```diff
@@ -1084,18 +1086,6 @@ mod test {
         assert_eq!(scores[usize::from(grm.nonterm_idx(&"E").unwrap())], 1);
     }
 
-    #[test]
```
Comment (ptersilie) [inline src/lib/yacc/grammar.rs:1087]: Why did you delete this test instead of just changing u64 to u32?

**Comment by ptersilie** [comment]
I agree with the above. `u32` is likely more than enough. Just one minor comment. Rest looks fine.

**Review by ptersilie — APPROVED** [review]

File: src/lib/yacc/grammar.rs (hunk 1 above)
Comment (ltratt) [reply src/lib/yacc/grammar.rs:1087]: The term costs are now `u8`: we'd need to have a grammar rule with over 16 million of them to trigger overflow. That seemed like too big for a test to bother worrying about ;)

Comment (ptersilie) [reply src/lib/yacc/grammar.rs:1087]: Ah, didn't see the `should_panic`. That's ok then.

**Comment by ptersilie** [comment]
I assume this can be merged as is?

**Comment by ltratt** [comment]
If you're happy with it, I am!

**Comment by ptersilie** [comment]
Merged.
